projects/serializers.py

```diff
@@ -86,7 +86,12 @@ class ProjectControlSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = ProjectControl
-        fields = ("status", "project", "control", "catalog_data", "component_data", )
+        fields = ("status", "project", "control", "remarks", "catalog_data", "component_data", )
+
+    def validate(self, attrs):
+        if attrs.get("status") == ProjectControl.Status.NA and attrs.get("remarks") is None:
+            raise serializers.ValidationError("A justification is required for non-applicable controls.")
```

Comment:
Need to also check that "remarks" isn't empty:
```suggestion
        if attrs.get("status") == ProjectControl.Status.NA and not attrs.get("remarks"):
            raise serializers.ValidationError("A justification is required for non-applicable controls.")
```

Reply:
@TMCamp Why not use this? Using `not attrs.get("status")` more concisely covers both the case where the value is an empty string or `None`.

Reply:
Found it. This is now a duplicate of a new conversation.